MapboxNavigation/NavigationViewController.swift

```diff
@@ -525,6 +535,14 @@ extension NavigationViewController: NavigationServiceDelegate {
         delegate?.navigationViewController(self, didFailToRerouteWith: error)
     }
     
+    public func navigationService(_ service: NavigationService, didRefresh routeProgress: RouteProgress) {
+        for component in navigationComponents {
+            component.navigationService(service, didRefresh: routeProgress)
+        }
+        
+        delegate?.navigationViewController(self, didRefresh: routeProgress)
```

Comment:
What about adding this functionality to `ViewController.swift`? I think it'd be beneficial for testing purposes to have it ready there (I can submit my changes if it makes sense as I did some testing to try it out).

Reply:
We can also add separate example in `navigation-ios-examples` to showcase this specific feature.

Reply:
Sure, feel free to push what you have.